main.sh

```diff
@@ -60,7 +60,24 @@ grep -f hg19.coding_subset.uniq1_2.bed hg19.coding_subset.sort1_2.bed > hg19.cod
 
 ## Task13: make a for loop to go through all bed files in the folder. For each bed file, count the no of transcripts on the negative strand. Save file names and corresponding counts in a single output file. Name the output file trans_negative.count
 
+#save following in trans_negative.count.sh file
+for bedfile in *.bed
+do
+	echo -n "$bedfile : "; awk '$6 ~ /-/ {n++} END {print n+0}' $bedfile
+done
+
+#in terminal
+./trans_negative.count.sh > trans_negative.count
 
 ## Task14: starting with the original input file, make a new column that has the length of coding sequence of each transcript. For example, a transcript that has 3 exons and their lengthes are 100,150,and 200, its length will be 100+150+200=450 bp. Save the new file under the name "hg19.coding_subset.coding_length.bed". Notice the difference from task no 5
+awk '{print $11}' hg19.coding_subset.bed | sed 's/,/\t/g' | awk '{for(i=1; i<=NF; i++) x+=$i; print x}' > length.bed 
```

Comment:
Perfect. BTW, you do not need to change "," into "\t" to use awk. You can use awk and change the input field separator.
awk '{print $11}' hg19.coding_subset.bed | awk -F"," '{for(i=1; i<=NF; i++) x+=$i; print x}' > length.bed